crates/vm/levm/src/opcode_handlers/block.rs

```diff
@@ -145,16 +146,35 @@ impl VM {
     }
 
     // BLOBHASH operation
+    /// Currently not tested
     pub fn op_blobhash(
         &mut self,
         current_call_frame: &mut CallFrame,
     ) -> Result<OpcodeSuccess, VMError> {
         self.increase_consumed_gas(current_call_frame, gas_cost::BLOBHASH)?;
 
-        // Should push in stack the blob hash
-        unimplemented!("when we have tx implemented");
+        let index = current_call_frame.stack.pop()?;
 
-        // Ok(OpcodeSuccess::Continue)
+        let index = index.as_usize();
+
+        let blob_hash: H256 = match &self.env.tx_blob_hashes {
+            Some(vec) => match vec.get(index) {
+                Some(el) => *el,
+                None => {
+                    return Err(VMError::TransactionDoesNotHaveABlobHashVector);
+                }
+            },
+            None => {
+                return Err(VMError::NotEnoughBlobHashes);
+            }
+        };
```

Comment:
I think that the errors are inverted here. The first match should return the error of "**missing blob hash vector**" and the internal match should return the "**blob hash not found**" error.
I would also rename these errors, some suggestions are:
**MissingBlobHashes** - when the vector is not found
**BlobHashIndexOutOfBounds** - when the blobhash is not found

Reply:
Done!